tests/test_google_oauth.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+import responses
+from allauth.socialaccount.models import SocialApp
+from django.contrib.auth import get_user_model
+from django.contrib.sites.models import Site
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIClient
+
+
+class TestGoogleAuth(TestCase):
+    """
+    Tests for Google Authentication !!
+    """
+
+    def setUp(self):
+        self.client = APIClient()
+        self.google_signup_url = reverse("google_login")
+        self.payload = {"access_token": "1234567", "id_token": "789456123"}
+        social_app = SocialApp.objects.create(
+            provider="google",
+            name="Google",
+            client_id="123456123456123456654321",
+            secret="654321123456123456987456123",
+        )
+        site = Site.objects.get_current()
+        social_app.sites.add(site)
+        
+    @responses.activate
+    def test_google_auth(self):
+        """
+        Testing Google Auth SignUp !!
+        """
+        self._get_google_response()
+        user_count = get_user_model().objects.all().count()
+
+        response = self.client.post(self.google_signup_url, data=self.payload, format="json")
```

Comment:
Check the payload @chinmaym07. Why are you posting access_token in the payload?
The payload must contain the thing that we pass from the frontend?